diff --git a/custom/dependencies/libpng/pngpriv.h b/custom/dependencies/libpng/pngpriv.h
--- a/custom/dependencies/libpng/pngpriv.h
+++ b/custom/dependencies/libpng/pngpriv.h
@@ -512,1 +512,1 @@
-    defined(THINK_C) || defined(__SC__) || defined(TARGET_OS_MAC)
+    defined(THINK_C) || defined(__SC__)
diff --git a/custom/dependencies/libzlib/zutil.h b/custom/dependencies/libzlib/zutil.h
--- a/custom/dependencies/libzlib/zutil.h
+++ b/custom/dependencies/libzlib/zutil.h
@@ -121,1 +121,1 @@
-#if defined(MACOS) || defined(TARGET_OS_MAC)
+#if defined(MACOS) /* Classic Mac only; iOS provides fdopen. */
